src/OnboardingSPA/pages/Steps/DesignHomepageMenu/index.js

```diff
@@ -76,41 +53,32 @@ const StepDesignHomepageMenu = () => {
 
 	function refactorPatterns( homepagePatternDataResp ) {
 		const makeHomepagePattern = [];
-
-		for ( const key in homepagesList ) {
-			const homepagePatterns = homepagesList[ key ];
-			// update the header menu pattern if already selected
-			if (
-				currentData.data.partHeader ||
-				currentData.data.partHeader !== ''
-			) {
-				homepagePatterns[ 0 ] = currentData.data.partHeader;
+		Object.values( homepagePatternDataResp ).forEach(
+			( homepagePatternData ) => {
+				makeHomepagePattern.push( homepagePatternData.content );
 			}
```

Comment:
Quick question, aren't lines 75-79 performing another `forEach`, is there anyway that we combine both of them? `setHomepagePattern` will be given an array of `contents` and `setHomepagePatternList` will be given an array of `slugs` both of which will be populated by a single `forEach` looping on `homepagePatternDataTemp?.body`